common/persistence/sql/sqlExecutionStoreUtil.go

```diff
@@ -422,6 +422,7 @@ func applyWorkflowMutationAsyncTx(
 	runID := serialization.MustParseUUID(executionInfo.RunID)
 
 	recoverPanic := func(err *error) {
+		// revive:disable-next-line:defer Func is being called using defer().
 		if r := recover(); r != nil {
```

Comment:
this is unfortunately not always true.  there are definitely some incorrect-and-not-working uses of `recoverPanic()` because Go is extremely picky about when and how `recover()` is called.

Reply:
All current uses of recoverPanic() are being called with defer() - I didn't mean to imply that you couldn't create future cases where you were using it wrong. The issue is that this lint warning doesn't alert at the call site, it alerts at the call to recover(), so it seems pretty useless right now. Do you have a better suggestion than ignoring it?

Reply:
ah, yeah, those look fine.

the one I was thinking of was this, which has apparently already been suppressed: 
https://github.com/uber/cadence/blob/8b42a6dccb725957b61c73dc10902f854c271ee2/common/log/panic.go#L35-L37

which is incorrect, it's used wrong in multiple places.  e.g. 
https://github.com/uber/cadence/blob/8b42a6dccb725957b61c73dc10902f854c271ee2/service/frontend/clusterRedirectionHandler.go#L1219-L1228

which is why I don't like hiding this lint, even though it's rather annoying in this file's correct usage.  it's an _extremely_ error-prone thing to do, and allowing it leaves room for both bad copy-pasta without understanding the details, and incorrect usage later without any warning.

Reply:
Right, but I don't think what we've got today is the solution. What we've got right now is a lint warning that everyone ignores, that doesn't point at line 1226 above, but rather just points at line 37 and does it regardless of the callers.

After reading and playing more with defer/recover, I'm certainly coming around to your point about not using recover-in-a-helper functions. In the case of CapturePanic(), that's 122 call sites to update and maintain. Ugh.

How about we add our own linter :P
`ag 'CapturePanic\(' | grep -v defer | wc -l
       1`

Reply:
because then we'd have
```go
func... () {
  defer afterCall(...)
}
func afterCall(...) {
  defer log.CapturePanic(..)
```
which is equally wrong :)

but yeah, 100% ugh but it's _so_ error-prone and _so_ hard to notice and _so_ potentially damaging that it's one where I think nearly any amount of pain is worth it.

I'm game to suppress it for now to enforce the other lints, and then follow up with restoring this one later / soon / as long as _it happens_, I'm happy.  I just fear forgetting it forever and then we encounter an easily-prevented outage.